client/integrations/integrations.go

```diff
@@ -804,6 +810,120 @@ func getVersionId(name string, filter string) (version string, err error) {
 	}
 }
 
+// ExportWithThreads exports all Integration Flows in the specified folder using a configurable number of threads
+func ExportWithThreads(folder string, numThreads int) error {
```

Comment:
You are not using `sync.WaitGroup`. Without it, how do we download 3 integrations in parallel (since 3 is the default) and wait for those responses before starting the next three.